noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/note.nr

```diff
@@ -0,0 +1,557 @@
+    /// interest of the executor of this fn to encrypt with random bytes.
+    log_bytes_padding_to_mult_31 = unsafe { get_random_bytes() };
+
+    let mut log_bytes = get_arr_of_size__log_bytes__from_PT::<(N * 32) + 64>();
+
+    assert(
+        log_bytes.len() % 31 == 0,
+        "Unexpected error: log_bytes.len() should be divisible by 31, by construction.",
+    );
+
+    log_bytes[0] = eph_pk_sign_byte;
+    let mut offset = 1;
+    for i in 0..header_ciphertext_bytes.len() {
+        log_bytes[offset + i] = header_ciphertext_bytes[i];
+    }
+    offset += header_ciphertext_bytes.len();
+
+    for i in 0..ciphertext_bytes.len() {
+        log_bytes[offset + i] = ciphertext_bytes[i];
+    }
+    offset += ciphertext_bytes.len();
+
+    for i in 0..log_bytes_padding_to_mult_31.len() {
+        log_bytes[offset + i] = log_bytes_padding_to_mult_31[i];
+    }
+
+    assert(
+        offset + log_bytes_padding_to_mult_31.len() == log_bytes.len(),
+        "Something has gone wrong",
+    );
+
+    // *****************************************************************************
+    // Convert bytes back to fields
+    // *****************************************************************************
+
+    let log_bytes_as_fields = be_bytes_31_to_fields(log_bytes);
+
+    // *****************************************************************************
+    // Prepend / append fields, to create the final log
+    // *****************************************************************************
+
+    // In this strategy, we prepend [tag, eph_pk.x]
+
+    // We assume that the sender wants for the recipient to find the tagged note,
+    // and therefore that they will cooperate and use the correct tag. Usage of a bad
+    // tag will result in the recipient not being able to find the note automatically.
+    let tag = unsafe { get_app_tag_as_sender(sender, recipient) };
+    increment_app_tagging_secret_index_as_sender(sender, recipient);
+
+    let mut final_log: [Field; PRIVATE_LOG_SIZE_IN_FIELDS] = [0; PRIVATE_LOG_SIZE_IN_FIELDS];
+
+    final_log[0] = tag;
+    final_log[1] = eph_pk.x;
+
+    let mut offset = 2;
+    for i in 0..log_bytes_as_fields.len() {
+        final_log[offset + i] = log_bytes_as_fields[i];
+    }
+    offset += log_bytes_as_fields.len();
+
+    for i in offset..PRIVATE_LOG_SIZE_IN_FIELDS {
+        /// Safety: randomness cannot be constrained.
+        final_log[i] = unsafe { random() };
+    }
+
+    final_log
+}
+
+unconstrained fn compute_log_unconstrained<Note, let N: u32>(
+    context: PrivateContext,
+    note: Note,
+    recipient: AztecAddress,
+    sender: AztecAddress,
+) -> [Field; PRIVATE_LOG_SIZE_IN_FIELDS]
+where
+    Note: NoteInterface<N>,
+{
+    compute_log(context, note, recipient, sender)
+}
+
+// This function seems to be affected by the following Noir bug:
+// https://github.com/noir-lang/noir/issues/5771
+// If you get weird behavior it might be because of it.
```

Comment:
This issue is closed, and we've also recently learned that the issue had to do with lambdas that take mutable args - it wasn't a bug (this: https://github.com/noir-lang/noir/pull/7127#issuecomment-2603210558)

Reply:
Removed the comment